docs/source/guides/build-options.rst

```diff
@@ -405,15 +405,26 @@ The following ``docker`` field contains some basic Docker configurations:
         - liblapack-dev
         - gfortran
 
+BentoML uses `BuildKit <https://github.com/moby/buildkit>`_, a cache-efficient builder toolkit, to containerize Bentos. BuildKit comes with `Docker 18.09 <https://docs.docker.com/develop/develop-images/build_enhancements/>`_. This means if you are using Docker via Docker Desktop, BuildKit will be available by default. If you are using a standalone version of Docker, you can install BuildKit by following the instructions `here <https://github.com/docker/buildx#installing>`_.
+
 .. note::
 
-   BentoML uses `BuildKit <https://github.com/moby/buildkit>`_, a cache-efficient builder toolkit, to containerize Bentos.
+    When using PyTorch or TensorFlow to run models on GPUs, you don't need to specify ``cuda_version`` in your ``bentofile.yaml`` to install the CUDA Toolkit separately.
```

Comment:
Installation of PyTorch and TensorFlow with CUDA using pip
==========================================================

In the past, it was commonly suggested to use the CUDA base image or install the correct CUDA version via `conda` when working with Nvidia GPUs alongside frameworks like PyTorch and TensorFlow. However, with Nvidia's recent release of CUDA as a PyPI package, those recommendations are now outdated.

Using the old method of installing CUDA separately can lead to:
- **Wasted disk space** due to installing redundant files.
- **Potential CUDA API version mismatches**, which can cause runtime errors.

The current recommended approach is to directly install both PyTorch and TensorFlow, along with their respective CUDA dependencies, via `pip`. This ensures:
- **Minimal package size** since only the required components are installed.
- **Better compatibility** as the correct CUDA version is automatically installed alongside the frameworks.

Installation Example
---------------------
To install PyTorch and TensorFlow with the appropriate CUDA version using pip, use the following commands:

.. code-block:: bash

    pip install torch
    pip install tensorflow[and-cuda]

These commands ensure that PyTorch, TensorFlow, and the correct version of CUDA are installed with minimal effort, providing a more efficient and reliable setup.

Reply:
Thanks. Too many details here, so I some of them to the pr.